### DISCUSSION TITLE
Does seaorm support nested structures?

### DISCUSSION THREAD

**Discussion by Raj2032** [SeaQL/sea-orm#2289] [General]
According to chat ai, it claims it does support it and provided this code:

```rust
use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "users")]
pub struct User {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
    pub address: Address,
}

#[derive(Clone, Debug, PartialEq)]
pub struct Address {
    pub street: String,
    pub city: String,
    pub state: String,
    pub zip: String,
}


let users = User::find()
    .filter(address.street.eq("123 Main St"))
    .all(db)
    .await?;
```

However I do not know if this is valid or not.

**Comment by seijikun** [comment]
I think this answers your question: https://github.com/SeaQL/sea-orm/discussions/2015

**Reply by Raj2032** [reply]
ok thanks :)

**Reply by seijikun** [reply]
Correction: It's actually currently being worked on in https://github.com/SeaQL/sea-orm/pull/2179 for queries.

**Reply by Raj2032** [reply]
@seijikun I see so I can query my nested structure similar to this:

```rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "users")]
pub struct User {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
    pub address: Address,
}

#[derive(Clone, Debug, PartialEq)]
pub struct Address {
    pub street: String,
    pub city: String,
    pub state: String,
    pub zip: String,
}
let users = User::find()
    .filter(address.street.eq("123 Main St"))
    .all(db)
    .await?;
```

after it has been worked on?

**Reply by seijikun** [reply]
@Raj2032 You will need extra structs for querying, it can't be your normal entity models (afaik).
And you will have to annotate the fields with nested models.
You can look at the new examples added by #2179 [>>here<<](https://github.com/SeaQL/sea-orm/blob/984827a6de82f965b41a1d7eb36852702eac8755/tests/partial_model_tests.rs#L194)

**Comment by Raj2032** [comment]
ok thanks mate :)
